RavenDB-12902 : AfterQueryExecuted is not called for aggregations and suggestions

aviv86 opened the pull request:
https://issues.hibernatingrhinos.com/issue/RavenDB-12902

aviv86 committed:
RavenDB-12902 : AfterQueryExecuted is not called for aggregations and suggestions

aviv86 committed:
RavenDB-12902 : tests

aviv86 commented:
need to check failing test
https://jenkins.hibernatingrhinos.com/job/PR_Tests_v4.2/1165/testReport/junit/SlowTests.Issues/RavenDB_9584/CanExtendAutoIndexWithSuggestions/

ppekrol commented on src/Raven.Client/Documents/Session/Operations/Lazy/LazySuggestionQueryOperation.cs:
```diff
@@ -56,7 +56,6 @@ public void HandleResponse(GetResponse response)
 
         private void HandleResponse(QueryResult queryResult)
         {
-            _invokeAfterQueryExecuted?.Invoke(queryResult);
```
When this event is invoked in lazy suggestion query now?

aviv86 replied:
in `ProcessResults`
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Client/Documents/Queries/Suggestions/SuggestionQuery.cs#L75

aviv86 committed:
RavenDB-12902 : fix failing tests -
when substituting provider.AfterQueryExecuted with documentQuery.AfterQueryExecuted,
we need to add the old provider.AfterQueryExecuted action to documentQuery
